src/sentry/snuba/discover.py

```diff
@@ -665,6 +727,9 @@ def timeseries_query(selected_columns, query, params, rollup, reference_event=No
     if len(snuba_args["aggregations"]) == 1:
         snuba_args["aggregations"][0][2] = "count"
 
+    if conditions is not None:
+        snuba_args["conditions"].extend(conditions)
```

Comment:
How do we know that aliases have been resolved on these conditions? It might be safer/better to add a `raw_query` call inside `key_transaction_timeseries_query`. That way we can ensure that aliases are always resolved correctly now and in the future.

Reply:
🤔 I can't think of a good way to do that without moving the rest of this function into a helper, what about changing the parameter to an optional `key_transaction_queryset` instead so its more explicit?

Reply:
I would likely duplicate the guts of this method personally. It is a small amount of duplication to avoid a potentially unsafe interface.

Reply:
Didn't end up duplicating, moved most of it to shared function instead. But can switch to duplication if that's not a good idea.